src/mrpro/operators/NonUniformFastFourierOp.py

```diff
@@ -0,0 +1,146 @@
+"""Non-Uniform Fast Fourier Operator."""
+
+from collections.abc import Sequence
+
+import numpy as np
+import torch
+from torchkbnufft import KbNufft, KbNufftAdjoint
+
+from mrpro.data.KTrajectory import KTrajectory
+from mrpro.operators.LinearOperator import LinearOperator
+
+
+class NonUniformFastFourierOp(LinearOperator):
```

Comment:
```suggestion
class NonUniformFastFourierOp(LinearOperator, adjoint_as_backward=True):
```
should fix torch.func while we continue to use torchkbnufft.